cmd/fleetctl/upgrade_packs_test.go

```diff
@@ -0,0 +1,393 @@
+	require.NoError(t, err)
+
+	got := runAppForTest(t, []string{"upgrade-packs", "-o", outputFile})
+	require.Contains(t, got, `Converted 0 queries from 2 2017 "Packs" into portable queries:`)
+	require.Contains(t, got, `visit https://example.com/packs/manage and disable all`)
+
+	content, err := os.ReadFile(outputFile)
+	require.NoError(t, err)
+	require.Empty(t, content)
+}
+
+func TestFleetctlUpgradePacks_NonEmpty(t *testing.T) {
+	_, ds := runServerWithMockedDS(t)
+
+	ds.UserByIDFunc = func(ctx context.Context, id uint) (*fleet.User, error) {
+		return &fleet.User{ID: id, GlobalRole: ptr.String(fleet.RoleAdmin)}, nil
+	}
+
+	ds.GetPackSpecsFunc = func(ctx context.Context) ([]*fleet.PackSpec, error) {
+		// queries must match those returned by ListScheduledQueriesInPackWithStats
+		return []*fleet.PackSpec{
+			{ID: 1, Name: "p1", Targets: fleet.PackSpecTargets{Labels: []string{"l1"}}, Queries: []fleet.PackSpecQuery{
+				{QueryName: "q1", Name: "sq1", Interval: 60, Snapshot: ptr.Bool(true), Platform: ptr.String("darwin")},
+			}},
+			{ID: 2, Name: "p2", Targets: fleet.PackSpecTargets{Teams: []string{"t1", "t2"}}, Queries: []fleet.PackSpecQuery{
+				{QueryName: "q2", Name: "sq2", Interval: 90, Removed: ptr.Bool(true), Platform: ptr.String("linux")},
+			}},
+		}, nil
+	}
+
+	ds.ListPacksFunc = func(ctx context.Context, opt fleet.PackListOptions) ([]*fleet.Pack, error) {
+		return []*fleet.Pack{
+			{ID: 1, Name: "p1", Labels: []fleet.Target{
+				{Type: fleet.TargetLabel, DisplayText: "l1"},
+			}, LabelIDs: []uint{1}},
+			{ID: 2, Name: "p2", Teams: []fleet.Target{
+				{Type: fleet.TargetTeam, DisplayText: "t1"},
+				{Type: fleet.TargetTeam, DisplayText: "t2"},
+			}, TeamIDs: []uint{1, 2}},
+		}, nil
+	}
+
+	ds.ListScheduledQueriesInPackWithStatsFunc = func(ctx context.Context, id uint, opts fleet.ListOptions) ([]*fleet.ScheduledQuery, error) {
+		// queries must match those returned by GetPackSpecs
+		if id == 1 {
+			return []*fleet.ScheduledQuery{
+				{ID: 1, PackID: id, Name: "sq1", QueryName: "q1", Interval: 60, Snapshot: ptr.Bool(true), Platform: ptr.String("darwin")},
+			}, nil
+		}
+		return []*fleet.ScheduledQuery{
+			{ID: 2, PackID: id, Name: "sq2", QueryName: "q2", Interval: 90, Removed: ptr.Bool(true), Platform: ptr.String("linux")},
+		}, nil
+	}
+
+	ds.CountHostsInTargetsFunc = func(ctx context.Context, filter fleet.TeamFilter, targets fleet.HostTargets, now time.Time) (fleet.TargetMetrics, error) {
+		return fleet.TargetMetrics{}, nil
+	}
+
+	ds.ListQueriesFunc = func(ctx context.Context, opt fleet.ListQueryOptions) ([]*fleet.Query, error) {
+		return []*fleet.Query{
+			{Name: "q1", Query: "select 1"},
+			{Name: "q2", Query: "select 2"},
+			{Name: "q3", Query: "select 3"},
+		}, nil
+	}
+
+	// expects a global query for p1 (targets a label) and per-team queries for p2 (t1 and t2)
+	b, err := os.ReadFile(filepath.Join("testdata", "expectedUpgradePacks.yml"))
+	require.NoError(t, err)
```

Comment:
For a reason I can't understand right now, this fails constantly on CI (but passes locally) with this error:

```
=== RUN   TestFleetctlUpgradePacks_NonEmpty
    upgrade_packs_test.go:326: 
        	Error Trace:	/home/runner/work/fleet/fleet/cmd/fleetctl/upgrade_packs_test.go:326
        	Error:      	Received unexpected error:
        	            	open testdata/expectedUpgradePacks.yml: no such file or directory
        	Test:       	TestFleetctlUpgradePacks_NonEmpty
--- FAIL: TestFleetctlUpgradePacks_NonEmpty (0.32s)
```

Of course the file is in this PR/branch/commit. This reads the content the same way that we do in e.g. `get_test.go` or `apply_test.go`. I'm at a loss, I will move the expected content as a string in the test, but if anyone has any idea (I did check file permissions and it is the same as the other testdata files AFAICS).